vignettes/function-flow.Rmd

```diff
@@ -97,113 +97,318 @@ library(dplyr)
+uses diagnoses registered in LPR2 and LPR3 to extract
+the dates of all recorded pregnancy endings (live births and
+miscarriages). These are identified by `diagnosis` values beginning with
+"DO0[0-6]", "DO8[0-4]" or "DZ3[37]". The dates output by
+`get_pregnancy_dates()` are used to exclude all inclusion events
+registered between 40 weeks before and 12 weeks after a pregnancy
+ending.
+
+After these exclusion functions have been applied, the output serves as
+inputs to two sets of functions:
+
+1.  The censored HbA1c and GLD data are passed to the
+    `join_inclusions()` function for the final step of the inclusion
+    process.
+2.  the censored GLD data is passed to the
+    `get_only_insulin_purchases()`,
+    `get_insulin_purchases_within_180_days()`, and
+    `get_insulin_is_two_thirds_of_gld_doses()` helper functions for the
+    classification of diabetes type.
+
+### Join inclusion events
+
+The function `join_inclusions()` appends/row-binds the dates output from
+functions the process the four types of inclusion events by `pnr`. Thus,
+it takes as input the following variables output from the following
+functions:
+
+-   From `include_diabetes_diagnoses()`:
+    -   `pnr`: identifier variable
+    -   `dates`: dates of the first and second hospital diabetes
+        diagnosis
+-   From `include_podiatrist_services()`
+    -   `pnr`: identifier variable
+    -   `dates`: the dates of the first and second diabetes-specific
+        podiatrist record
+-   From `exclude_pregnancy()`:
+    -   `pnr`: identifier variable
+    -   `dates`: the dates of the first and second elevated HbA1c test
+        results (after censoring)
+-   From `exclude_pregnancy()`:
+    -   `pnr`: identifier variable
+    -   `date`: dates of all purchases of GLD
+        -   The dates of the first and second purchase of GLD of each
+            individual are extracted from these and appended as two rows
+            to the ´dates´ variable.
+
+The output from the function is a `data.frame` containing two variables
+(`pnr` and `dates`) and 1 to 8 rows per ´pnr´. This output is passed to
+`get_diagnosis_date()`.
 
 ### Get diagnosis date
 
-The function `get_diagnosis_date()` combines the outputs from the
-inclusion and exclusion functions to get the final diagnosis date.
-Initially, it drops the first inclusion and exclusion events from the
-function outputs with the helper `drop_first_event()`, so that only
-those with two or more events are kept. This is then used to assign an
-initial diagnosis according to OSDC. Then, all the outputs are joined
-together with `join_diagnosis_dates()`.
-
-Finally, the dates outside of the data coverage period are dropped with
-`drop_diagnosis_dates_outside_coverage()` to end with a final diagnosis
-date. For details on this censoring based on periods with insufficient
-data coverage, see the `vignette("design")`.
+The function `get_diagnosis_date()` takes the output from
```

Comment:
Per https://github.com/steno-aarhus/osdc/pull/133#discussion_r1891349171 , we might want to rename this function:

```suggestion
The function `get_inclusion_date()` takes the output from
```